### PR TITLE
Refactor editor to mvp

### LINKED ISSUE #48: Refactor editor to use MVP
The editor app doesn't use the MVP (model view presenter) pattern. 
As the app get bigger it will benefit from the MVP pattern

1. Create a model and presenter for the app in main.py
2. Ensure all the widgets communicate with the model via the presenter

### PR DESCRIPTION
**Pull request by TimofeiZavileiskii**
* **Please check if the PR fulfills these requirements**
- [x] The commit message follows our guidelines
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

Close #48

### CODE REVIEW

**Comment by TimofeiZavileiskii** [comment]
Does every function in the view must have a docstring or if it is small enough it is fine?

File: sscanss/editor/widgets.py [hunk 1]
```diff
@@ -319,7 +319,7 @@ def __init__(self, parent):
         super().__init__(parent)
         self.parent = parent
 
-        self.instrument = self.parent.instrument
+        self.instrument = parent.instrument
```
Comment (StephenNneji) [inline sscanss/editor/widgets.py:322]: This should be `parent.presenter.model.instrument`. Please ensure all the widgets and dialogs are accessing the instrument correctly

File: sscanss/editor/view.py [hunk 2]
```diff
@@ -0,0 +1,289 @@
...
+        self.message.setReadOnly(True)
+        self.message.setFontFamily('courier')
+        self.message.setStyleSheet('QTextEdit {background-color: white; color: red; font-size: 12px; '
+                                   'padding-left: 10px; background-position: top left}')
+        self.message.setMinimumHeight(100)
+        self.main_splitter.addWidget(self.message)
+
+        self.gl_widget = OpenGLRenderer(self)
+        self.gl_widget.custom_error_handler = self.sceneSizeErrorHandler
+        self.scene = SceneManager(self, self.gl_widget, False)
```
Comment (StephenNneji) [inline sscanss/editor/view.py:41]: should be `SceneManager(self.presenter.model, self.gl_widget, False)`

File: sscanss/editor/presenter.py [hunk 3]
```diff
@@ -0,0 +1,240 @@
...
+        self.updateTitle()
+
+    def setInstrumentSuccess(self, result):
+        """Sets the instrument created from the instrument file.
+
+        :param result: instrument from description file
+        :type result: Instrument
+        """
+        self.view.setMessageText("OK")
+        self.view.setInstrument(result)
```
Comment (StephenNneji) [inline sscanss/editor/presenter.py]: should be `self.model.instrument = result`

**Pull request merged**